File: outco/curriculum/target_practice_prompts/go/w4_d1_oop.go
package main

import (
	"fmt"
)

type Board struct {
  // YOUR WORK HERE
}


func (board *Board) Init(numRows int, numCols int) *Board {
  // YOUR WORK HERE
  return board
}


func (board *Board) printBoard() {
    // YOUR WORK HERE
}

func (board *Board) canPlacePiece(row int, col int) bool {
    // YOUR WORK HERE
    return false
}

func (board *Board) placePiece(row int, col int, player int) {
    // YOUR WORK HERE
}

func (board *Board) checkWinCondtion(player int) bool {
    // YOUR WORK HERE
    return false
}

func (board *Board) checkDiagonals(player int) bool {
    // YOUR WORK HERE
    return false
}

func (board *Board) checkRows(player int) bool {
    // YOUR WORK HERE
    return false
}

func (board *Board) checkColumns(player int) bool {
    // YOUR WORK HERE
    return false
}


type TicTacToe struct {
  // YOUR WORK HERE
}

func (ttt *TicTacToe) Init() *TicTacToe {
  // YOUR WORK HERE
  return ttt
}

func (ttt *TicTacToe) play() {
  // YOUR WORK HERE
}

func (ttt *TicTacToe) playRound(row int, col int) {
  // YOUR WORK HERE
}


func (ttt *TicTacToe) printCurrentPlayersTurn() {
  // YOUR WORK HERE
}

func (ttt *TicTacToe) decrementRounds() {
  // YOUR WORK HERE
}

func (ttt *TicTacToe) declareWinner() {
  // YOUR WORK HERE
}

func (ttt *TicTacToe) declareTie() {
  // YOUR WORK HERE
}

func (ttt *TicTacToe) printInvalidMove(){
  // YOUR WORK HERE
}

func (ttt *TicTacToe) printCurrentMove(row int, col int) {
  // YOUR WORK HERE
}

func (ttt *TicTacToe) switchPlayer() {
  // YOUR WORK HERE
}


func main() {
  fmt.Println("Let's Play!")
  // YOUR WORK HERE
}

File: outco/curriculum/target_practice_prompts/swift/w4_d1_oop.swift

class TicTacToe {

  init() {
    // YOUR WORK HERE
    }

  func play() {
    // YOUR WORK HERE
  }

  func playRound() {
      // YOUR WORK HERE
  }

  func printCurrentPlayersTurn() {
      // YOUR WORK HERE
  }

  func decrementRounds() {
      // YOUR WORK HERE
  }

  func declareWinner(_ play: Character) {
    // YOUR WORK HERE
  }

  func declareTie() {
    // YOUR WORK HERE
  }

  func printInvalidMove() {
    // YOUR WORK HERE
  }

  func printCurrentMove() {
    // YOUR WORK HERE
  }

  func switchPlayer() {
    // YOUR WORK HERE
  }
}

class Board {

  init() {
    // YOUR WORK HERE
  }

  func printBoard() {
    // YOUR WORK HERE
  }

  func canPlacePiece(_ x: Int, _ y: Int) -> Bool {
    // YOUR WORK HERE
    return false 
  }

  func placePiece(_ x: Int, _ y: Int) {
    // YOUR WORK HERE
  }

  func checkWinCondition(for play: Character) -> Bool {
    // YOUR WORK HERE
    return false
  }

  func checkDiagnols(for play: Character) -> Bool {
    // YOUR WORK HERE
    return false
  }

  func checkRows(for play: Character) -> Bool {
    // YOUR WORK HERE
    return false
  }

  func checkColumns(for play: Character) -> Bool {
    // YOUR WORK HERE
    return false
  }
}

let ttt = TicTacToe()
ttt.play()
